curp/src/server/raw_curp/mod.rs

```diff
@@ -489,6 +495,11 @@ impl<C: 'static + Command, RC: RoleChange + 'static> RawCurp<C, RC> {
             return Err(st_w.term);
         }
         if term > st_w.term {
+            let timeout = st_w.follower_timeout_ticks;
+            if st_w.leader_id.is_some() && self.ctx.election_tick.load(Ordering::Acquire) < timeout
```

Comment:
We already check that the leader is None in `PreVote` phase, as the entire process is `PreVote` -> `Vote`, should we check it again here?

Reply:
Removed